### PR TITLE
account_default_if_zero_lamport always returns a value

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
moving to not mmapping append vec files. Soon, lifetimes of borrowed account data will be limited to a callback.

#### Summary of Changes
Rework `StorableAccounts`.`account_default_if_zero_lamport()` to always return a value.
This simplifies the api and allows a callback to work better when the data's lifetime is limited. Life time changes are coming soon.

Fixes #

### CODE REVIEW

File: accounts-db/src/storable_accounts.rs [hunk 1]
```diff
@@ -90,10 +96,14 @@ pub trait StorableAccounts<'a>: Sync {
     fn account_default_if_zero_lamport<Ret>(
         &self,
         index: usize,
-        mut callback: impl FnMut(Option<AccountForStorage<'a>>) -> Ret,
+        mut callback: impl FnMut(AccountForStorage<'a>) -> Ret,
```
Comment (brooksprumo) [inline accounts-db/src/storable_accounts.rs:99]: Once this change goes in, I think it'll be valuable to consolidate `account()` and `account_default_if_zero_lamport()` into a single method. Supporting both of these is likely unnecessary, and removing one will make code easier to reason about IMO.

I'd love to get to a point where we don't need to special case how we handle `lamports == 0` here, and instead can guarantee callers always handle `lamports == 0` properly.

File: accounts-db/src/storable_accounts.rs [hunk 2]
```diff
@@ -75,6 +75,12 @@ impl<'a> ReadableAccount for AccountForStorage<'a> {
     }
 }
 
+lazy_static! {
+    static ref DEFAULT_ACCOUNT_SHARED_DATA: AccountSharedData = AccountSharedData::default();
+    pub static ref DEFAULT_ACCOUNT: AccountForStorage<'static> =
```
Comment (brooksprumo) [inline accounts-db/src/storable_accounts.rs:80]: Does this need to be `pub`? I haven't used lazy_static enough to know off the top of my head. Is `DEFAULT_ACCOUNT` used outside of this file?

Comment (jeffwashington) [reply accounts-db/src/storable_accounts.rs:80]: you're right. it doesn't have to be pub. The next pr changes this lazy static block again. I'll remove that pub in there. There are several changes stacked behind this one.

Comment (jeffwashington) [reply accounts-db/src/storable_accounts.rs:80]: Actually, the `pub static ref DEFAULT_ACCOUNT...` line entirely goes away in the next pr. So, this is temporary.

File: accounts-db/src/storable_accounts.rs (hunk 1 above)
Comment (jeffwashington) [reply accounts-db/src/storable_accounts.rs:99]: sounds good. I would have to look into the details of why we want this or don't. Maybe we always want to set to default?

**Review by brooksprumo — APPROVED** [review]
:shipit:

**Pull request merged**